### DISCUSSION TITLE
Will you support Solidity language?

### DISCUSSION THREAD

**Discussion by Confucian-e** [massCodeIO/massCode#180] [Q&A]
Hello guys, I'm a blockchain developer and also your honest fan.
Will you support solidity language? 
That's important for blockchain developers.

Thank you and good luck!

**Comment by antonreshetov** [comment]
The answer is here https://github.com/massCodeIO/massCode/discussions/170

**Comment by Confucian-e** [comment]
Well it's hard for me to make PR but I have found that `.tmlanguage` here https://github.com/davesag/Solidity.tmbundle/blob/develop/Syntaxes/Solidity.tmLanguage
Could you please add solidity syntax highlights?

**Comment by antonreshetov** [comment]
I tried it, but both with Mikrotik and Solidity the conversion tool crashed with an error. There is an issue on this subject https://github.com/ajaxorg/ace/issues/4604

Ace editor is pretty old and the developers are not supporting it as well as before. 

I have a plan to replace the editor with some other one. But the problem with all editors is that they have their own tokenizer which is different from `.tmLanguage` grammar. This is the main problem of additional syntax support. I am looking for a solution that can just use `.tmLanguage` without additional manipulation.

**Reply by Confucian-e** [reply]
Thank you guy. I'm so appreciate.

**Reply by antonreshetov** [reply]
@Confucian-e  [v3.0.0-beta.1](https://github.com/massCodeIO/massCode/releases/tag/v3.0.0-beta.1) supports Solidity

![image](https://user-images.githubusercontent.com/17480987/183029707-1e83d4fc-a98b-4d55-ad2c-d2668421d2b1.png)

**Reply by Confucian-e** [reply]
Awesome. It's important for blockchain developers! Thank you guys.
